sim/shaman/lightning_bolt.go

```diff
@@ -38,25 +38,29 @@ func (shaman *Shaman) newLightningBoltSpell(isLightningOverload bool) *core.Spel
 	}
 
 	if shaman.HasMajorGlyph(proto.ShamanMajorGlyph_GlyphOfLightningBolt) {
-		spellConfig.DamageMultiplier *= 1.04
+		spellConfig.DamageMultiplier += 0.04
 	}
 
 	if shaman.HasSetBonus(ItemSetSkyshatterRegalia, 4) {
-		spellConfig.DamageMultiplier *= 1.05
+		spellConfig.DamageMultiplier += 0.05
 	}
 
-	applyDot := !isLightningOverload && shaman.HasSetBonus(ItemSetWorldbreakerGarb, 4)
 	var lbDot *core.Dot
-	lbdotDmg := 0.0 // dynamically changing dmg
-	lbDotSpell := shaman.RegisterSpell(core.SpellConfig{
-		ActionID:         core.ActionID{SpellID: 64930},
-		SpellSchool:      core.SpellSchoolNature,
-		ProcMask:         core.ProcMaskEmpty,
-		Flags:            core.SpellFlagIgnoreModifiers,
-		DamageMultiplier: 1,
-		ThreatMultiplier: 1,
-	})
-	if applyDot {
+	if !isLightningOverload && shaman.HasSetBonus(ItemSetWorldbreakerGarb, 4) {
+		lbDotSpell := shaman.RegisterSpell(core.SpellConfig{
+			ActionID:         core.ActionID{SpellID: 64930},
+			SpellSchool:      core.SpellSchoolNature,
+			ProcMask:         core.ProcMaskEmpty,
+			Flags:            core.SpellFlagNoOnCastComplete | core.SpellFlagIgnoreModifiers,
+			DamageMultiplier: 1,
+			ThreatMultiplier: 1,
+
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				lbDot.ApplyOrReset(sim)
+				spell.CalcAndDealOutcome(sim, target, spell.OutcomeAlwaysHit)
```

Comment:
I had it so that this was outside the cast so that way you didn't have to call cast unless the dot wasn't ticking... I guess this way at least you see the exact number of times it was actually applied. I just liked how clean the timeline looked haha

Reply:
This is mostly for uniformity, so all rolling DOTs are implemented (and look) the same way